File: Java/closestDivisible/closestDivisible.java
package closestDivisible;
class Main {
    static int closestNumber(int n, int m) {
        // find the quotient
        int closest = 0;
        int minDifference = Integer.MAX_VALUE;

        // Check numbers around n
        for (int i = n - Math.abs(m); i <= n + Math.abs(m); ++i) {
            if (i % m == 0) {
                int difference = Math.abs(n - i);

                if (difference < minDifference || 
                   (difference == minDifference && Math.abs(i) > Math.abs(closest))) {
                    closest = i;
                    minDifference = difference;
                }
            }
        }
        return closest;
    }

    public static void main(String[] args) {
        int n = 13, m = 4;
        System.out.println(closestNumber(n, m));
    }
}

File: LogicBuilding/closestDivisible.js
function closestNumber(n, m) {

    let q = parseInt(n / m);
    
    let n1 = m * q;
    
    let n2 = (n * m) > 0 ?
        (m * (q + 1)) : (m * (q - 1));
    
    if (Math.abs(n - n1) < Math.abs(n - n2))
        return n1;
    
    return n2;
}

// Driver Code
let n = 13;
let m = 4;
console.log(closestNumber(n, m));